### PR TITLE
feat(JOML): cleanup and migrate code to joml

### PR DESCRIPTION
**Pull request by pollend**
launch game with AdventureAssets and verify these blocks act correctly with these changes:
- give AltarOfResurrectionRoot
- give FireballLauncherRoot
- give SwingingBladeRoot
- give WipeOutRoot

### CODE REVIEW

**Comment by jdrueckert** [comment]
blocked by / dependent on joml migration of `StructureTemplates`

File: src/main/java/org/terasology/adventureassets/traps/fireballlauncher/FireballLauncherServerSystem.java [hunk 1]
```diff
@@ -30,7 +31,7 @@
 import org.terasology.entitySystem.systems.UpdateSubscriberSystem;
 import org.terasology.logic.health.HealthComponent;
 import org.terasology.logic.location.LocationComponent;
-import org.terasology.math.geom.Vector3f;
+import org.terasology.math.JomlUtil;
```
Comment (jdrueckert) [inline src/main/java/org/terasology/adventureassets/traps/fireballlauncher/FireballLauncherServerSystem.java]: Is this still needed?

File: src/main/java/org/terasology/adventureassets/traps/swingingblade/SwingingBladeSettingsScreen.java [hunk 2]
```diff
@@ -83,10 +86,10 @@ private void onSaveButton(UIWidget button) {
             double yawValue = Math.toRadians(Double.parseDouble(yaw.getText()));
             double pitchValue = Math.toRadians(Double.parseDouble(pitch.getText()));
             double rollValue = Math.toRadians(Double.parseDouble(roll.getText()));
-            locationComponent.setWorldRotation(new Quat4f((float) yawValue, (float) pitchValue, (float) rollValue));
+            locationComponent.setWorldRotation(new Quaternionf().rotationYXZ((float) yawValue, (float) pitchValue, (float) rollValue));
```
Comment (jdrueckert) [inline src/main/java/org/terasology/adventureassets/traps/swingingblade/SwingingBladeSettingsScreen.java:89]: why `rotationYXZ`?

File: src/main/java/org/terasology/adventureassets/traps/swingingblade/SwingingBladeUtilities.java [hunk 3]
```diff
@@ -25,13 +26,14 @@ static void rotateSwingingBlade(EntityRef blade, float gameTime) {
...
-            float pitch, A = swingingBladeComponent.amplitude, phi = swingingBladeComponent.offset;
+            float pitch;
+            float a = swingingBladeComponent.amplitude;
+            float phi = swingingBladeComponent.offset;
             float w = (float) (2 * Math.PI / timePeriod);
-            pitch = (float) (A * Math.cos(w * t + phi));
+            pitch = (float) (a * Math.cos(w * gameTime + phi));
             Quat4f rotation = locationComponent.getLocalRotation();
-            locationComponent.setLocalRotation(new Quat4f(rotation.getYaw(), pitch, rotation.getRoll()));
+            locationComponent.setLocalRotation(new Quaternionf().rotationYXZ(rotation.getYaw(), pitch, rotation.getRoll()));
```
Comment (jdrueckert) [inline src/main/java/org/terasology/adventureassets/traps/swingingblade/SwingingBladeUtilities.java:36]: Same question as above

Comment (pollend) [reply src/main/java/org/terasology/adventureassets/traps/swingingblade/SwingingBladeUtilities.java:36]: new Qua4f(float,float,float) is the same as new Quaternionf().rotationYXZ(yaw,pitch,roll). the axises in terasology are flipped for Y and X for rotation. there are some test cases I have with this and I believe I wrote out and compared the two at some point.

File: src/main/java/org/terasology/adventureassets/traps/fireballlauncher/FireballLauncherServerSystem.java (hunk 1 above)
Comment (jdrueckert) [inline src/main/java/org/terasology/adventureassets/traps/fireballlauncher/FireballLauncherServerSystem.java]:
```suggestion
```

**Review by jdrueckert — APPROVED** [review]

**Pull request merged**